hack/update-vendor-licenses.sh

```diff
@@ -197,14 +197,14 @@ for PACKAGE in $(go list -m -json all | jq -r .Path | sort -f); do
     continue
   fi
   # Skip a directory if 1) it has no files and 2) all the subdirectories contain a go.mod file.
-  if [[ -z "$(find "${DEPS_DIR}/${PACKAGE}" -mindepth 1 -maxdepth 1 -type f)" ]]; then
+  if [[ -z "$(find "${DEPS_DIR}/${PACKAGE}/" -mindepth 1 -maxdepth 1 -type f)" ]]; then
       misses_go_mod=false
       while read -d "" -r SUBDIR; do
           if [[ ! -e "${SUBDIR}/go.mod" ]]; then
               misses_go_mod=true
               break
           fi
-      done < <(find "${DEPS_DIR}/${PACKAGE}" -mindepth 1 -maxdepth 1 -type d -print0)
+      done < <(find "${DEPS_DIR}/${PACKAGE}/" -mindepth 1 -maxdepth 1 -type d -print0)
```

Comment:
A heads up that this does technically change the value of `SUBDIR` in the loop (although in an inconsequential way) because `find` will *always* append a `/` to the given path: https://github.com/kubernetes/kubernetes/pull/92735/files#r448662042